wrangler/src/txFunctions/runCB.js

```diff
@@ -0,0 +1,197 @@
+import { response, Stopwatch } from 'cfw-easy-utils'
+import { Transaction, Networks, Keypair } from 'stellar-base'
+import BigNumber from 'bignumber.js'
+import moment from 'moment'
+import { find as loFind } from 'lodash'
+
+import txSponsorsSettle from '../txSponsors/settleCB'
+
+// TODO:
+  // Remove all old txSponsor stuff
+
+export default async ({ event, request, params }) => {
+  const { txFunctionHash } = params
+
+  const { value, metadata } = await TX_FUNCTIONS.getWithMetadata(txFunctionHash, 'arrayBuffer')
+
+  if (!value)
+    throw {status: 404, message: `txFunction could not be found this turret`}
+
+  const { length, txFunctionSignerPublicKey, txFunctionSignerSecret } = metadata
+
+  const txFunctionBuffer = Buffer.from(value)
+  const txFunction = txFunctionBuffer.slice(0, length).toString()
+
+  const body = await request.json()
+  const claimableBalanceToken = request.headers.get('authorization')?.split(' ')?.[1]
+
+  if (!claimableBalanceToken)
+    throw {message: `txFunctionFee is missing`}
+
+  const [
+    claimableBalanceId, 
+    claimableBalanceSignature, 
+    ...txFunctionHashes
+  ] = JSON.parse(Buffer.from(claimableBalanceToken, 'base64'))
+
+  if (
+    txFunctionHashes.length
+    && txFunctionHashes.indexOf(txFunctionHash) === -1
+  ) throw {message: `txFunctionFee is invalid for this txFunction`}
+
+  const { metadata: feeMetadata } = await TX_FEES.getWithMetadata(claimableBalanceId)
+
+  let feeTotalBigNumber
+  let feeSpentBigNumber
+
+  if (feeMetadata) {
+    feeTotalBigNumber = new BigNumber(feeMetadata.total)
+    feeSpentBigNumber = new BigNumber(feeMetadata.spent)
+
+    if (feeSpentBigNumber.isGreaterThanOrEqualTo(feeTotalBigNumber)) {
+      event.waitUntil(txSponsorsSettle(claimableBalanceId))
```

Comment:
The idea here is to claim once some1 attempts to use a balance that is larger than the total we consume it? Then, the user for the turret is responsible to update a new claimable balance, make a token, and make the same call again?

Do we intend to keep any fee bump methodology in place? Or is that now not possible? I worry slightly about random crons and what not having to manage this., or for time-sensitive contract execution.

Reply:
> The idea here is to claim once some1 attempts to use a balance that is larger than the total we consume it? Then, the user for the turret is responsible to update a new claimable balance, make a token, and make the same call again?
That's right

> Do we intend to keep any fee bump methodology in place? Or is that now not possible? I worry slightly about random crons and what not having to manage this., or for time-sensitive contract execution.
Not sure what fee bumps would buy us anymore, but yeah, they're no longer in play. If you have a time-sensitive contract that needs executing don't use a token that's nearly spent.